pkg/invoke/delegate.go

```diff
@@ -15,13 +15,52 @@
 package invoke
 
 import (
+	"encoding/json"
 	"fmt"
 	"os"
 	"strings"
 
 	"github.com/containernetworking/cni/pkg/types"
 )
 
+func GetNextPlugin(netconf []byte) (string, []byte, error) {
+	conf := struct {
+		types.NetConf
+		Inner map[string]interface{} `json:"inner,omitempty"`
+	}{}
+
+	err := json.Unmarshal(netconf, &conf)
+	if err != nil {
+		return "", nil, fmt.Errorf("error decoding net config: %v", err)
+	}
+
+	if conf.Inner != nil {
+		tmp, ok := conf.Inner["type"]
+		if !ok {
+			return "", nil, fmt.Errorf("inner plugin type not specified")
+		}
+		innerType, ok := tmp.(string)
+		if !ok {
+			return "", nil, fmt.Errorf("inner plugin type not a string")
+		}
+
+		// Copy network name and CNI version to inner plugin
+		conf.Inner["name"] = conf.Name
+		conf.Inner["cniVersion"] = conf.CNIVersion
```

Comment:
We should be setting the `args` field for the inner conf, by copying the `args` field from the outer conf.